### PR TITLE
Added embree parameter support

### PR DESCRIPTION
**Pull request by Luvideria**
This PR adds parsing of several embree parameters.
Embree exposes two functions:

```
rtcSetSceneFlags(RTCScene,` RTCSceneFlags)
rtcSetSceneBuildQuality(RTCScene,RTCBuildQuality)
```

The implementation uses to_json and from_json implemented for RTCSceneFlags and RTCBuildArguments.

Currently supported:

- Build quality
- - 0: morton codes
- - 1: binned sah
- - 2: triangle split binned sah
- scene flags
- - compact
- - dynamic
- - robust

If Embree exposes an API to modify all the RTCBuildArguments, the interface is ready, otherwise we need to implement it ourselves in Embree such as in https://github.com/Luvideria/embree3-AT
using
`rtc_AT_SetNextBVHArguments(RTCScene, RTCBuildArguments);`

example usage:
```
lm.load_accel("accel","embree",{"quality":0, "compact":1})
``` 
Defaut values are:
quality: 1
compact: 0 / False
dynamic: 0 / False
robust: 0 / False

### CODE REVIEW

File: plugin/accel_embree/CMakeLists.txt [hunk 1]
```diff
@@ -22,6 +22,7 @@ if (embree_FOUND)
         SOURCES
             "embree.h"
             "accel_embree.cpp"
+            "embree.cpp"
```
Comment (hi2p-perim) [inline plugin/accel_embree/CMakeLists.txt]: The file name of embree.h might be confusing with official header. you can use something more explicit like embree_params.h

File: plugin/accel_embree/accel_embree.cpp [hunk 2]
```diff
@@ -33,9 +33,28 @@ class Accel_Embree final : public Accel {
...
+        Json j;
+        to_json(j,settings_);
+        LM_INFO(j.dump());
+        j.clear();
+        
+        to_json(j,sf_);
+        LM_INFO( j.dump() );
+        
+        }
+
```
Comment (hi2p-perim) [inline plugin/accel_embree/accel_embree.cpp]: to_json and from_json functions can be implemented with nlohman_json's adl_serialier. ref: `https://github.com/nlohmann/json`

File: plugin/accel_embree/embree.cpp [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#include "embree.h"
+#include <lm/json.h>
+LM_NAMESPACE_BEGIN(LM_NAMESPACE)
+
+#define tryAt(j,str,v) try{j.at(str).get_to(v);}catch(...){ v=v;}
```
Comment (hi2p-perim) [inline plugin/accel_embree/embree.cpp]: tryAt macro can be replaced with lm::json::value functions, which can accept a default value.

File: plugin/accel_embree/embree.cpp [hunk 4]
```diff
@@ -0,0 +1,66 @@
+#include "embree.h"
```
Comment (hi2p-perim) [inline plugin/accel_embree/embree.cpp]: I think embree.cpp is simple enough to merge into embree.h

**Comment by Luvideria** [comment]
New version implement requested changes and uses  `=` operator instead of `to_json` and `from_json`

**Pull request merged**